app/models/test_track/application_identity.rb

```diff
@@ -0,0 +1,32 @@
+class TestTrack::ApplicationIdentity
+  include Singleton
+
+  delegate :test_track_ab, to: :private_identity
+
+  private
+
+  def app_name
+    raise 'must configure TestTrack.app_name on application initialization' if TestTrack.app_name.blank?
+    TestTrack.app_name
+  end
+
+  def private_identity
+    PrivateIdentity.new(app_name)
+  end
+
+  class PrivateIdentity
```

Comment:
i added this to further hide the rest of the `TestTrack::Identity` api that doesn't make sense in the ApplicationIdentity context, such as `test_track_login!` and `test_track_vary`. I'm not sure it's worth it, because TestTrack::ApplicationIdentity is already hidden behind the `TestTrack` module. Any opinions here?

Reply:
This feels kinda weird to me - I don't think I would have been able to figure out what's going on if you didn't leave this comment here. If we only want to have test_track_ab here can we just copy those 3 lines from `TestTrack::Identity`?

Reply:
spoke offline, i think copying the three methods from TestTrack::Identity does not achieve the api that we want (we'd have to expose two additional public methods), and i think it involves duplication of somewhat complex code. I prefer using an approach that exposes only the minimal api that we need, and makes use of our existing patterns for managing visitor session

Reply:
but i'd be more open to a refactor or another pattern of using that that may be more self explanatory? these names feel a bit off, but they are all hidden behind a simpler public api, so they don't seem super important either

Reply:
I don't love this name and i don't have a good sense for a better pattern to use right now. Fortunately, this is all private API, with the exception of `TestTrack::ApplicationIdentity` which i think is a good name.